x-pack/plugins/siem/public/management/pages/policy/store/policy_details/selectors.ts

```diff
@@ -31,22 +34,24 @@ export const policyDetailsForUpdate: (
 
 /** Returns a boolean of whether the user is on the policy details page or not */
 export const isOnPolicyDetailsPage = (state: Immutable<PolicyDetailsState>) => {
-  if (state.location) {
-    const pathnameParts = state.location.pathname.split('/');
-    return pathnameParts[1] === 'policy' && pathnameParts[2];
-  } else {
-    return false;
-  }
+  return (
+    matchPath(state.location?.pathname ?? '', {
+      path: MANAGEMENT_ROUTING_POLICY_DETAILS_PATH,
+      exact: true,
+    }) !== null
+  );
 };
 
 /** Returns the policyId from the url */
 export const policyIdFromParams: (state: Immutable<PolicyDetailsState>) => string = createSelector(
   state => state.location,
   (location: PolicyDetailsState['location']) => {
-    if (location) {
-      return location.pathname.split('/')[2];
-    }
-    return '';
+    return (
+      matchPath<ManagementRoutePolicyDetailsParams>(location?.pathname ?? '', {
```

Comment:
That react router utility also extracts our the route params - so no more custom parsing logic